File: javascript/sieve/sieve.js
export const primes = (limit) => {
    let sieve = Array(limit + 1).fill(true);

    sieve[0] = false;
    sieve[1] = false;

    for (let i = 2; i <= limit; i++) {
        if (sieve[i]) {
            for (let j = i + i; j <= limit; j += i) {
                sieve[j] = false;
            }
        }
    }

    return sieve.reduce((acc, prime, index) => prime ? [...acc, index] : acc, []);
};

File: python/sieve/sieve.py
def primes(limit):
    if limit < 2:
        return []

    sieve = [True]*(limit + 1)
    sieve[0] = False
    sieve[1] = False

    res = []

    for i in range(2, limit + 1):
        if sieve[i]:
            res.append(i)
            for j in range(2 * i, limit + 1, i):
                sieve[j] = False

    return res
